### PR TITLE
[terra-embedded-content-consumer] Add title prop to component so that title can be easily set into the frame

### PR DESCRIPTION
**Pull request by kolkheang**
### Summary

Add title prop to the component so that it's face up and easier to set the title of the content in the frame

**What was changed:**
Add title prop to the component so consumer can easily set the title of the iframe

**Why it was changed:**
For better accessibility, having a title text of the content in the frame is critical. This change is to make setting the title easier.

### Testing

This change was tested using:

- [ ] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [x] Other (please describe below)
- [ ] No tests are needed

The change can also be tested by inspecting the source code by looking for the `title` attribute for the `<iframe>` using the examples site: https://engineering.cerner.com/terra-framework/pull/1776.

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [x] Functional review

### Additional Details

VoiceOver announcing the title of the frame:
https://github.com/cerner/terra-framework/assets/3170314/01c0088d-9dbf-4255-95ae-f697350dc482

JAWS announcing the title of the frame:
https://github.com/cerner/terra-framework/assets/3170314/47ad7d1f-168a-4a73-8c11-104344a34b11

**This PR resolves:**

UXPLATFORM-9419 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (RayGunY) [inline packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:48]: Nit: could you put a newline between this const and the if statement

File: packages/terra-framework-docs/CHANGELOG.md
Comment (RayGunY) [inline packages/terra-framework-docs/CHANGELOG.md:6]: nit: add a period

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (RayGunY) [inline packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:49]: This section of code should be able to be moved into the if statement directly below it.

Talked with Kol and it looks like we only care about this check/set if we are calling into the `if (this.props.title)` statement. If we don't call into that statement then we can just keep `frameOptions.iframeAttrs == null` and don't need to set it to `frameOptions.iframeAttrs = {};`.

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (chenyixuan625) [inline packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:51]: Could we use `===` instead of `==`? Same for L57

**Comment by RayGunY** [comment]
Overall I'm happy with this, could you post a recording of this in edge with JAWs? After that I'm good to approve this.

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kolkheang) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:51]: It's different. I am using `==` to check for `null` or `undefined`. I can adjust to include a check for `undefined` as well to make it clear.

**Comment by kolkheang** [comment]
> Overall I'm happy with this, could you post a recording of this in edge with JAWs? After that I'm good to approve this.

https://github.com/cerner/terra-framework/assets/3170314/47ad7d1f-168a-4a73-8c11-104344a34b11

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kolkheang) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx]: I updated the check here:
https://github.com/cerner/terra-framework/pull/1776/files#diff-77b1c1b39ad95249bd3956ef509a0995b4687f8c94ba87a3110f3e6b1ce8ccfeR51-R59

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kolkheang) [inline packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:59]: @chenyixuan625  - I updated the check here

**Review by chenyixuan625 — APPROVED** [review]

**Review by sycombs — APPROVED** [review]

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (supreethmr) [inline packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:16]: I think this should be rename to `titleText` OR `iframeTitle` because in the past we had renamed prop `title` with `text` for terra header components as it would conflict with HTML `title` attribute.

**Comment by sdadn** [comment]
Same comment as: https://github.com/cerner/terra-framework/pull/1777#issuecomment-1717756734

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kenk2) [inline packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:50]:
```suggestion
    frameOptions.iframeAttrs = this.props.title && frameOptions.iframeAttrs ? frameOptions.iframeAttrs " {}
```

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kenk2) [inline packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:56]: Similar comment about why we're doing this deleting of untruthy keys: Is this really necessary?

**Review by kenk2 — CHANGES_REQUESTED** [review]

**Review by scottwilmarth — APPROVED** [review]

**Review by eawww — APPROVED** [review]
Works as expected

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (scottwilmarth) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:16]: @supreethmr this is the HTML title attribute that I am asking to be added. Without it, iframes will not pass accessibility requirements.

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (scottwilmarth) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:50]: @sdadn and all, **for an iFrame to be accessible it must have the title attribute.** By including "title" as a first class prop we can implement measures to ensure the title is provided and eventually make it required. As is, it is not guaranteed that any engineer will add it. Yes we can beef up our documentation, but that also requires people to read which they do not regularly do. I am specifically asking for this prop to be added to ensure we help enforce best practices. While we have to make it a passive change for now, I will push to make it required as soon as possible.

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kolkheang) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:16]: @supreethmr - this prop is for the EmbeddedContentConsumer component which will be passed down to the actual iframe attribute. I don't see any conflict with it.

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kolkheang) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:56]: My initial thought was that we don't want any key without a corresponding value set. But, I could see consumers wanting to have some key set specifically with null or undefined.

I can remove that check.

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kolkheang) [inline packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:50]: @kenk2 - thanks for the suggestion:
```
frameOptions.iframeAttrs = this.props.title && frameOptions.iframeAttrs ? frameOptions.iframeAttrs : {}
```
I'll rework the logic a bit.

**Comment by kolkheang** [comment]
@sdadn  / @supreethmr  / @kenk2 / @sycombs 

Please re-review, and approve

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kolkheang) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:51]: I re-worked the logic a bit in https://github.com/cerner/terra-framework/pull/1776/commits/3ffd8305ee593960880ba6b3ac9bbdab84932e6c

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kolkheang) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:52]: I removed that check to allow more flexibility for the consumers.
https://github.com/cerner/terra-framework/pull/1776/commits/3ffd8305ee593960880ba6b3ac9bbdab84932e6c

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kolkheang) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:51]: Re-worked the logic with https://github.com/cerner/terra-framework/pull/1776/commits/3ffd8305ee593960880ba6b3ac9bbdab84932e6c

**Review by kenk2 — APPROVED** [review]

**Comment by sdadn** [comment]
After an offline discussion, we landed on the following to to unblock and merge this PR:

1. Throw a warning if `options.iframeAttrs.title` is being set
2. Update documentation to discourage the use of `options.iframeAttrs.title` and to use `title` instead
3. Add a test to make sure `title` overrides `options.iframeAttrs.title`
4. Add TODO comments

      - to remove the warning in the next MVB
      - to set `title` as required in the next MVB

**Comment by kolkheang** [comment]
@sdadn  / @sycombs  / @RayGunY  / @chenyixuan625 / @scottwilmarth 

Please review and approve

File: packages/terra-framework-docs/src/terra-dev-site/doc/embedded-content-consumer/example/BasicConsumer.jsx
Comment (RayGunY) [inline packages/terra-framework-docs/src/terra-dev-site/doc/embedded-content-consumer/example/BasicConsumer.jsx:11]: Do you still need to add options={{}} here?

Comment (kolkheang) [reply packages/terra-framework-docs/src/terra-dev-site/doc/embedded-content-consumer/example/BasicConsumer.jsx:11]: No, it's not needed. That will be undefined but the code in `EmbeddedContentConsumer` will check and add an empty object.

Comment (kolkheang) [reply packages/terra-framework-docs/src/terra-dev-site/doc/embedded-content-consumer/example/BasicConsumer.jsx:11]: It's optional as well.

**Review by RayGunY — APPROVED** [review]

**Review by chenyixuan625 — APPROVED** [review]

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kolkheang) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:16]: No. This comment will not show in the doc site.
> // TODO MVB - set `title` prop as required

File: packages/terra-embedded-content-consumer/tests/jest/EmbeddedContentConsumer.test.jsx
Comment (kolkheang) [reply packages/terra-embedded-content-consumer/tests/jest/EmbeddedContentConsumer.test.jsx:10]: Updated

File: packages/terra-embedded-content-consumer/CHANGELOG.md
Comment (kolkheang) [reply packages/terra-embedded-content-consumer/CHANGELOG.md:6]: Updated the message

File: packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx
Comment (kolkheang) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:82]: Yes, it only displays when options.iframeAttrs.title is set.

Comment (kolkheang) [reply packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx:82]: Screenshot 2023-09-28 at 6 40 21 PM
Screenshot 2023-09-28 at 6 41 31 PM

**Review by sdadn — APPROVED** [review]

**Pull request merged**
